lib/ArtifactoryAPI.js

```diff
@@ -271,4 +272,39 @@ ArtifactoryAPI.prototype.downloadFile = function (repoKey, remotefilePath, desti
   return deferred.promise;
 };
 
+/** Upload Build Information
+ * @param   {object} buildInfo - see build.json {@link https://www.jfrog.com/confluence/display/RTF/Artifactory+REST+API#ArtifactoryRESTAPI-BuildUpload} {@link https://github.com/JFrogDev/build-info#build-info-json-format}
+ * @returns {object} A QPromise to a string containing the result.
+ */
+ArtifactoryAPI.prototype.uploadBuild = function (buildInfo) {
+  var deferred = Q.defer();
+  if(!buildInfo.name || !buildInfo.number) {
+    deferred.reject('Build Info must include a name and number. See https://www.jfrog.com/confluence/display/RTF/Artifactory+REST+API#ArtifactoryRESTAPI-BuildUpload for more info');
+  }

```

Comment:
You need to return the deferred object here, otherwise the other lines below this will be called (particularly request.put which I am sure you don't want to call.

`
deferred.reject('reason');
return deferred.promise;
`